packages/js/email-editor/src/components/block-editor/editor.tsx

```diff
@@ -32,6 +32,12 @@ export function InnerEditor( {
 	initialEdits,
 	...props
 } ) {
+	const { loadPersonalizationTags } = useDispatch( storeName );
+
+	useEffect( () => {
+		void loadPersonalizationTags();
+	}, [ loadPersonalizationTags ] );
```

Comment:
I'm not sure I understand why we have this here. Is it an optimization to preload personalization tags, or is it the main mechanism for fetching them into the app? I don't see any other place where we fetch so I assume it is the latter. The wordpress/data has resolvers which are meant exactly for this see [the docs](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#:~:text=normalized%20object.-,resolvers,-A%20resolver%20is)

We don't have any resolver in our store, so you will need to add the first one.  I was looking into docs and asked Chat GPT (`How to add a resolver to store made using @wordpress/data`), and it looks pretty straightforward.

Reply:
Good point with a resolver. I refactored this initialization into a proposed solution.